### PR TITLE
ShortTermRetentionManagedInstance

### PR DESCRIPTION
**Pull request by v-djnisi**
## Description

This are clients for short term retention on managed instance. I generated new clients and added new scenario tests. 

Azure-Rest-API-Specs pull request: https://github.com/Azure/azure-rest-api-specs/pull/3979
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@v-djnisi please join aka.ms/azuregithub to get access to the azure org and get authenticated to kick off CI process.

**Comment by v-djnisi** [comment]
@shahabhijeet Done

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 		<PackageId>Microsoft.Azure.Management.Sql</PackageId>
 		<Description>Azure SQL Management SDK library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-		<Version>1.21.0-preview</Version>
+		<Version>1.22.0-preview</Version>
 		<PackageTags>Microsoft Azure SQL Management;SQL;SQL Management;</PackageTags>
 		<PackageReleaseNotes>
 			<![CDATA[
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Please update `PackageReleaseNotes`

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs [hunk 2]
```diff
@@ -94,16 +95,5 @@ internal static partial class SdkInfo
               }.AsEnumerable();
           }
```
Comment (dsgouda) [inline src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: Please pull latest remote changes. run `msbuild build.proj` and regenerate the code

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part

**Comment by dsgouda** [comment]
@v-djnisi please join Azure org [here](http://aka.ms/azuregithub) to kick off the build

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj (hunk 1 above)
Comment (v-djnisi) [reply src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: Done

File: src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs (hunk 2 above)
Comment (v-djnisi) [reply src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs]: Done

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good, please squash commits into a single commit and we should be good to merge

**Comment by v-djnisi** [comment]
@dsgouda This is the first time I am using git. I synced the code midway, and I have no idea how to to squash commits, since they are not consecutive, and are already pushed. I tried rebasing, but couldn't get it to work properly. Can you help me with this?

**Comment by dsgouda** [comment]
There is a simple way to do it.
1. Copy all the files you have modified in a temp location outside the repo
2. Checkout psSdkJson6 branch, pull latest changes from remote (git pullhttps://github.com/Azure/azure-sdk-for-net.git psSdkJson6)
3. Create a new branch (git checkout -b branch_name)
4. Copy over the changes from temp location to this branch
5. Push this branch to remote (git push upstream branch_name)
6. Close this PR and open a new PR with the new branch and reference this PR there

Hope this helps

**Comment by v-djnisi** [comment]
@dsgouda Thank you for the help. This is the new PR https://github.com/Azure/azure-sdk-for-net/pull/4982.

**Pull request closed without merging**